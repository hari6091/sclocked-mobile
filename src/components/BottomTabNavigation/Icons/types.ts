import { SvgProps } from 'react-native-svg';

export interface IconProps extends SvgProps {
  focused: boolean;
  color?: string;
  size?: number;
}
