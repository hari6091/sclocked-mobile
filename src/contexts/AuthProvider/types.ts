interface ILogin {
  email: string;
  password: string;
}

export interface IUser {
  email?: string;
  username?: string;
  token?: string;
}

export interface IContext extends IUser {
  signIn: ({ email, password }: ILogin) => Promise<void>;
  logout: () => void;
}

export interface IAuthProvider {
  children: JSX.Element;
}
