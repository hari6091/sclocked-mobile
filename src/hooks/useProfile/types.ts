export interface MyUser {
  uuid?: string;
  id?: string;
  name: string;
  tags: string;
  matricula: string;
  email: string;
  disciplinaOUcargo: string;
  role: string;
  createdAt: string;
}

export interface MyUserSalas {
  uuid?: string;
  id?: string;
  name: string;
  numero: string;
  grupo: string;
  status: string;
  users: MyUser[];
}

export type IUserUpdate = {
  id?: string | undefined;
  name: string | undefined;
  tags: string | undefined;
  matricula: string | undefined;
  disciplinaOUcargo: string | undefined;
  email: string | undefined;
  password: string;
  confPassword: string;
  role: string | undefined;
};
