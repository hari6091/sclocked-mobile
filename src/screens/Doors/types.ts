import { screens } from "../../constants";
import { MainTabScreenProps } from "../../routes/types";

export type DoorsScreenProps = MainTabScreenProps<screens.DOOR>;

export interface ISalaOpt {
  salaNumber: string;
  status: string;
}
