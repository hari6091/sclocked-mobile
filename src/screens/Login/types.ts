import { screens } from "../../constants";
import { RootStackScreenProps } from "../../routes/types";

export type LoginScreenProps = RootStackScreenProps<screens.LOGIN>;

export type SignInType = {
  email: string;
  password: string;
};
