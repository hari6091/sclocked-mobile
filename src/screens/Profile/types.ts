import { screens } from "../../constants";
import { MainTabScreenProps } from "../../routes/types";

export type ProfileScreenProps = MainTabScreenProps<screens.PROFILE>;
