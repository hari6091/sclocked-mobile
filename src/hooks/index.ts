export * from "./useAuth";
export * from "./useProfile";
export * from "./useSalaAuth";
