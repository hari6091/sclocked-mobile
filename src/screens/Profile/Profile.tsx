import { Center, Text, Button, VStack } from "native-base";
import React from "react";
import { ProfileScreenProps } from "./types";
import { screens } from "../../constants";
import { useAuth, useProfile } from "../../hooks";
import { UserHeader } from "../../components";

export function dataCadastro(param: string | undefined) {
  const data = new Date(param ?? "");
  const dia = data.getDate().toString();
  const diaF = dia.length === 1 ? `0${dia}` : dia;
  const mes = (data.getMonth() + 1).toString();
  const mesF = mes.length === 1 ? `0${mes}` : mes;
  const anoF = data.getFullYear();
  return `${diaF}/${mesF}/${anoF}`;
}

const Profile = ({ navigation }: ProfileScreenProps) => {
  const { logout } = useAuth();
  const { profile } = useProfile();

  const handleNavigateLogin = async () => {
    await logout();
    navigation.navigate(screens.LOGIN);
  };

  return (
    <>
      <UserHeader />
      <Center
        flex={1}
        flexDirection="column"
        safeArea
        px="10"
        bg="#FFFFFF"
        pb="42px"
      >
        <VStack
          w="100%"
          bg="#28374D"
          borderRadius="20px"
          py="22px"
          px="16px"
          space="12px"
          justifyContent="center"
        >
          <Text color="#FFFFFF" fontSize="32px" fontWeight={700}>
            Perfil
          </Text>
          <Text color="#FFFFFF" fontSize="24px" fontWeight={700}>
            Nome do Usuário
          </Text>
          <Text color="#FFFFFF" fontSize="24px" testID="profile-username">
            {profile?.name ?? "Carregando..."}
          </Text>
          <Text color="#FFFFFF" fontSize="24px" fontWeight={700}>
            Cargo / Nível de acesso
          </Text>
          <Text color="#FFFFFF" fontSize="24px" testID="profile-cargo">
            {profile?.disciplinaOUcargo ?? "Carregando..."} /{" "}
            {profile?.role === "admin" ? "Administrador" : "Usuário comum"}
          </Text>
          <Text color="#FFFFFF" fontSize="24px" fontWeight={700}>
            Ingresso no Sistema
          </Text>
          <Text color="#FFFFFF" fontSize="24px" testID="profile-data-criado">
            {dataCadastro(profile?.createdAt)}
          </Text>
          <Button
            w="full"
            bg="#2486CE"
            _text={{ fontWeight: 700, fontSize: "24px" }}
            onPress={handleNavigateLogin}
            testID="profile-logout-button"
          >
            Sair
          </Button>
        </VStack>
      </Center>
    </>
  );
};

export default Profile;
