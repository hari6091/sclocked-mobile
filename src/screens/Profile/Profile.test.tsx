import React from "react";
import Profile from "./Profile";
import { render, fireEvent, act } from "../../test/test-utils";
import { ProfileScreenProps } from "./types";
import { useAuth, useProfile } from "../../hooks";
import { screens } from "../../constants";

jest.mock("../../hooks/useAuth");
jest.mock("../../hooks/useProfile");

describe("Profile screen", () => {
  const navigation = {
    setOptions: jest.fn(),
    navigate: jest.fn(),
  } as unknown as ProfileScreenProps["navigation"];

  const route = {
    params: undefined,
  } as unknown as ProfileScreenProps["route"];

  beforeAll(() => {
    const logoutMock = jest.fn();
    const mockProfile = {
      name: "John Doe",
      disciplinaOUcargo: "Professor",
      role: "admin",
      createdAt: "2023-07-29T12:34:56",
    };

    (useAuth as jest.Mock).mockReturnValue({
      logout: logoutMock,
    });

    (useProfile as jest.Mock).mockReturnValue({
      profile: mockProfile,
    });
  });

  it("renders screen, and email and password fields correctly", () => {
    const { getByTestId, toJSON } = render(
      <Profile navigation={navigation} route={route} />
    );

    expect(getByTestId("profile-username")).toBeTruthy();
    expect(getByTestId("profile-cargo")).toBeTruthy();

    expect(getByTestId("profile-data-criado")).toBeTruthy();
    expect(getByTestId("profile-logout-button")).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders user's name correctly", () => {
    const { getByTestId } = render(
      <Profile navigation={navigation} route={route} />
    );
    expect(getByTestId("profile-username")).toHaveTextContent("John Doe");
  });

  it("renders user's role correctly", () => {
    const { getByTestId } = render(
      <Profile navigation={navigation} route={route} />
    );
    expect(getByTestId("profile-cargo")).toHaveTextContent(
      "Professor / Administrador"
    );
  });

  it("renders user's creation date correctly", () => {
    const { getByTestId } = render(
      <Profile navigation={navigation} route={route} />
    );
    expect(getByTestId("profile-data-criado")).toHaveTextContent("29/07/2023");
  });

  it("calls logout function when 'Sair' button is pressed", () => {
    const { getByTestId } = render(
      <Profile navigation={navigation} route={route} />
    );

    const logoutButton = getByTestId("profile-logout-button");

    fireEvent.press(logoutButton);

    expect(useAuth().logout).toHaveBeenCalledTimes(1);
  });
});
