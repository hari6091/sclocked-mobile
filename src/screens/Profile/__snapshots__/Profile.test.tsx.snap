// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile screen renders screen, and email and password fields correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": "17%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text />
    <View
      dataSet={{}}
      style={
        {
          "width": 22,
        }
      }
    />
    <View
      dataSet={{}}
      style={{}}
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#28374D",
            "fontFamily": undefined,
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 27,
            "textAlign": "center",
            "textDecorationLine": undefined,
          }
        }
        testID="username"
      >
        John Doe
      </Text>
    </View>
  </View>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingBottom": 42,
        "paddingLeft": 40,
        "paddingRight": 40,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "backgroundColor": "#28374D",
          "borderRadius": 20,
          "justifyContent": "center",
          "paddingBottom": 22,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 22,
          "width": "100%",
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 32,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 48,
            "textDecorationLine": undefined,
          }
        }
      >
        Perfil
      </Text>
      <View
        dataSet={{}}
        style={
          {
            "height": 12,
          }
        }
      />
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
      >
        Nome do Usuário
      </Text>
      <View
        dataSet={{}}
        style={
          {
            "height": 12,
          }
        }
      />
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
        testID="profile-username"
      >
        John Doe
      </Text>
      <View
        dataSet={{}}
        style={
          {
            "height": 12,
          }
        }
      />
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
      >
        Cargo / Nível de acesso
      </Text>
      <View
        dataSet={{}}
        style={
          {
            "height": 12,
          }
        }
      />
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
        testID="profile-cargo"
      >
        Professor
         /
         
        Administrador
      </Text>
      <View
        dataSet={{}}
        style={
          {
            "height": 12,
          }
        }
      />
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
      >
        Ingresso no Sistema
      </Text>
      <View
        dataSet={{}}
        style={
          {
            "height": 12,
          }
        }
      />
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
        testID="profile-data-criado"
      >
        29/07/2023
      </Text>
      <View
        dataSet={{}}
        style={
          {
            "height": 12,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#2486CE",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 10,
            "width": "100%",
          }
        }
        testID="profile-logout-button"
      >
        <View
          dataSet={{}}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          test={true}
        >
          <View
            dataSet={{}}
            style={{}}
          >
            <Text
              dataSet={{}}
              style={
                {
                  "backgroundColor": undefined,
                  "color": "#fafafa",
                  "fontFamily": undefined,
                  "fontSize": 24,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "letterSpacing": 0,
                  "lineHeight": 36,
                  "textDecorationLine": undefined,
                }
              }
            >
              Sair
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
